core/src/main/resources/xml/xsl/adapter2mermaid.xsl

```diff
@@ -0,0 +1,552 @@
+					<xsl:with-param name="originalPipes">
+						<xsl:copy-of select="$newOriginalPipes"/>
+					</xsl:with-param>
+					<xsl:with-param name="iterationDepth" select="$iterationDepth+1"/>
+				</xsl:apply-templates>
+			</xsl:for-each>
+		</xsl:variable>
+
+		<!-- When processing the pipes via forwards, loops might result in pipes being duplicated, so deduplicate them and only keep the earliest occurence -->
+		<xsl:choose>
+			<xsl:when test="$finalProcessing">
+<!--				<test1>-->
+				<xsl:for-each-group select="$processedPipes/*" group-by="@elementID">
+					<xsl:variable name="sorted">
+						<xsl:for-each select="current-group()">
+							<xsl:sort select="@errorHandling"/>
+							<xsl:copy-of select="."/>
+						</xsl:for-each>
+					</xsl:variable>
+<!--					<xsl:sort select="@iterationDepth"/>-->
+					<xsl:copy-of select="$sorted/*[1]"/>
+				</xsl:for-each-group>
+<!--				</test1>-->
+<!--				<test>-->
+<!--					<xsl:for-each select="$processedPipes/*">-->
+<!--						<xsl:sort select="@errorHandling"/>-->
+<!--						<xsl:sort select="@iterationDepth"/>-->
+<!--						<xsl:copy-of select="."/>-->
+<!--					</xsl:for-each>-->
+<!--				</test>-->
+
+				<!-- The process of following forwards skips orphaned pipes, but still show those TBD -->
+
+			</xsl:when>
+			<xsl:otherwise><xsl:copy-of select="$processedPipes"/></xsl:otherwise>
+		</xsl:choose>
+	</xsl:template>
+
+	<xsl:template name="defaultCopyActions">
+		<xsl:attribute name="elementID" select="generate-id()"/>
+		<xsl:apply-templates select="*|@*" mode="#current"/>
+	</xsl:template>
+	
+	<xsl:template name="defaultPipeCopyActions">
+		<xsl:call-template name="defaultCopyActions"/>
+		<xsl:apply-templates select="../global-forwards/forward[not(@name = current()/forward/@name)]" mode="#current"/>
+	</xsl:template>
+	
+	<xsl:template name="createForward">
+		<xsl:param name="path"/>
+		<xsl:param name="name" select="$path"/>
+		
+		<xsl:variable name="forward">
+			<xsl:element name="forward">
+				<xsl:attribute name="name" select="$name"/>
+				<xsl:attribute name="path" select="$path"/>
+			</xsl:element> 
+		</xsl:variable>
+		<xsl:apply-templates select="$forward" mode="#current"/>
+	</xsl:template>
+
+	<xsl:template match="*|@*|text()" mode="convertElements">
+		<xsl:apply-templates select="*|@*|text()" mode="#current"/>
+	</xsl:template>
+
+	<xsl:template match="adapter" mode="convertElements">
+		<xsl:if test="number($adapterCount) gt 1">
+			<xsl:text>	style </xsl:text>
+			<xsl:value-of select="@name"/>
+			<xsl:text disable-output-escaping="yes"> fill:#0000,stroke:#1a9496,stroke-width:2px&#10;</xsl:text>
+			<xsl:text>	subgraph </xsl:text>
+			<xsl:value-of select="@name"/>
+			<xsl:text>&#10;</xsl:text>
+		</xsl:if>
+		<xsl:apply-templates select="*|@*|text()" mode="convertElements"/>
+		<xsl:if test="number($adapterCount) gt 1">
+			<xsl:text>	end&#10;</xsl:text>
+		</xsl:if>
+	</xsl:template>
+
+	<xsl:template match="receiver" mode="convertElements">
```

Comment:
Alle templates die hierna volgen, hebben eigenlijk dezelfde output-opbouw, kan je daar niet een gemeenschappelijke template voor maken die dat doet waarbij alleen maar parameters meegegeven hoeven te worden die werkelijk anders zijn in plaats  van iedere keer het hele blok kopieren?

Reply:
Ik heb dit geimplementeerd en ik het ziet er inderdaad beter uit.